R/cdisc_data.R

```diff
@@ -33,29 +33,28 @@
 #'   )
 #' )
 cdisc_data <- function(...,
-                       join_keys = teal.data::join_keys(),
+                       join_keys = teal.data::cdisc_join_keys(...),
                        code = "",
                        check = FALSE) {
   data_objects <- list(...)
+  deprecated_join_keys_extract(data_objects, join_keys)
+  teal_data(..., join_keys = join_keys, code = code, check = check)
+}
 
-  # todo: is it really important? - to remove
-  if (inherits(join_keys, "JoinKeySet")) {
-    join_keys <- teal.data::join_keys(join_keys)
-  }
-
-  if (
-    checkmate::test_list(data_objects, types = c("TealDataConnector", "TealDataset", "TealDatasetConnector"))
-  ) {
-    lifecycle::deprecate_warn(
-      when = "0.3.1",
-      "cdisc_data(
-        data_objects = 'should use data directly. Using TealDatasetConnector and TealDataset is deprecated.'
-      )"
-    )
-    update_join_keys_to_primary(data_objects, join_keys)
+#' Extrapolate parents from `TealData` classes
+#'
+#' note: This function will be removed once the following classes are defunct:
+#'  `TealDataConnector`, `TealDataset`, `TealDatasetConnector`
+#'
+#' @keywords internal
+deprecated_join_keys_extract <- function(data_objects, join_keys) {
```

Comment:
You need to add this condition to skip `deprecated_join_keys_extract` in case you provide named list of objects into `...`

```suggestion
deprecated_join_keys_extract <- function(data_objects, join_keys) {
  if (checkmate::test_list(data_objects, types = c("TealDataConnector", "TealDataset", "TealDatasetConnector"))) {
    return(NULL)
  }
```

Reply:
Thanks! I noticed this too when testing. Added on 9630699 _(negated though)_